sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -502,6 +502,6 @@ class IncrementalExecution(
       case p: StateStoreWriter => p.shouldRunAnotherBatch(
         watermarkPropagator.getInputWatermarkForEviction(tentativeBatchId,
           p.stateInfo.get.operatorId))
-    }.exists(_ == true)
+    }.contains(true)
```

Comment:
nit: unrelated change? Please file a new minor PR.

Reply:
will remove these minor changes from this PR.